### PR TITLE
Multi label Data and MultiLabel Accuracy

### PR DESCRIPTION
**Pull request by sguada**
This PR allows to use multi-label data {-1,0,1} with 0 means ignore, -1 negative and 1 positive.

@sergeyk review

### CODE REVIEW

**Comment by tangyuxing** [comment]
This pull request contains merge conflicts that must be resolved?

**Comment by bhack** [comment]
@tangyuxing This was superseded by @netheril96 PRs

**Comment by bgreen-litl** [comment]
Is https://github.com/BVLC/caffe/pull/1414 really intended to replace this PR or is it IndirectionLayer just one part of it?  (Trying to get my bearings in these two very heavily commented PRs.)

**Comment by bhack** [comment]
@bgreen-litl There is also https://github.com/BVLC/caffe/pull/1380

**Comment by elaksana** [comment]
Has there been any progress on this PR? @sguada Have you been working lately on this?

**Comment by rajeswar18** [comment]
@sguada I am trying to learn a model that outputs a word(more than 10 characters) so how to write the code length as a binary vector? should it be [0,..9,>9] or could you let me kknow the current trend for multi-label

**Comment by danielhauagge** [comment]
I'm really interested in this PR or any other one that implements it's functionality, so far #1380 and #1414 seem to be working on the same problem.

@sguada, if I were to try and do some multilabel learning with caffe right now, which pr would you recommend? If any.

**Comment by gavinmh** [comment]
I am also interested in data and accuracy layers that support multi-label classification. Can someone clarify the states of the PRs?

**Comment by bhack** [comment]
@gavinmh @danielhauagge See  last commenta in https://github.com/BVLC/caffe/pull/1380

**Comment by SoldierNeverDie** [comment]
fatal error: cublas_v2.h: No such file or directory
same problem, did you deal with it? @misaghkord 
or anybody can help me? THX!

**Comment by barbolo** [comment]
Wow! Tomorrow is exactly a year since this PR was created.

If I need to solve a multi label classification problem, what is the approach I should follow with caffe?

ps: I'm still digging through all the references posted here.

**Comment by bhack** [comment]
Is this only a documentation problem now? See https://github.com/BVLC/caffe/issues/1698

**Comment by iamhankai** [comment]
Where can I find "window_file_2007_trainval.txt" in "pascal_finetune_trainval_test.prototxt" (https://github.com/BVLC/caffe/blob/master/examples/finetune_pascal_detection/pascal_finetune_trainval_test.prototxt)?

**Comment by shelhamer** [comment]
Closing this PR since it is against the defunct dev branch and we concluded that losses and layers are capable of handling multilabel problems so this is a documentation / example issue and not a technical problem that needs further layers.

**Comment by ih4cku** [comment]
The dev branch has been deprecated and deleted. Why didn't this feature merged into the master branch?

**Comment by bhack** [comment]
@ih4cku It is not needed. See https://github.com/BVLC/caffe/issues/1698

**Comment by ih4cku** [comment]
@bhack 
Thanks for responding. I'm doing the same task as [Goodfellow's multi-digit problem](http://arxiv.org/abs/1312.6082v4), which you also mentioned. 
In this multi-task setting, it's ok for training by using the multi-dimensional labels with `SigmoidCrossEntropyLayer` or slicing the labels to feed to  multiple `SoftmaxWithLoss` layers. 
But how about the accuracy? How to compute the total accuracy of a sample during test? 
AFAIK, now the `AccuracyLayer` only support the target label blob of shape (N ×1×1×1), which is mentioned in this issue #2188.
Have you customized the accuracy layer?

**Comment by bhack** [comment]
I've done something cited in comments at https://github.com/BVLC/caffe/pull/2935  but was never reviewed and now closed because against dev branch.

**Comment by urakagi** [comment]
Where is the document/example explaining how to handle multilabel problems with current system?

**Comment by ColdZoo** [comment]
Many thanks to @sguada , I am using your code now.

**Comment by Coderx7** [comment]
Is this finalized?

**Comment by sidmangla** [comment]
Hi there, siddharth here, i am working on a project to recognize products and have used 200 pics each to train the 11 classes. but when i am testing it is showing only one class at a time. For labeling i have used LabelImg.
please help
